Packages/com.unity.inputsystem/InputSystem/Plugins/UI/InputSystemUIInputModule.cs

```diff
@@ -1660,13 +1660,28 @@ private int GetPointerStateIndexFor(InputControl control, bool createIfNotExists
                 touchId = touchscreen.primaryTouch.touchId.ReadValue();
                 touchPosition = touchscreen.primaryTouch.position.ReadValue();
             }
+
+            int displayIndex = 0;
+#if UNITY_INPUT_SYSTEM_ENABLE_MULTIDISPLAY && (UNITY_EDITOR_WIN || UNITY_EDITOR_OSX || UNITY_EDITOR_LINUX || UNITY_ANDROID)
```

Comment:
I just confirmed with Joel, this change does work in editor and android only. So this logic should be correct 👍

Reply:
I don't know of any method (short of adding that #define to the top of every file) of doing that.  Do you know a way to define that symbol conditionally and globally?

Reply:
Sorry for back to my old question:
So then I should read the PR description about 'There is also no implementation that properly passes this index over for any of the Editor or Android build targets' as the behaviour _before_ this PR lands?

That implies that 'Currently, this property only being set in certain Player build targets that support multidisplay' is currently available today, before this PR lands.

If so, why is there a special path here for editors/android? And not just using whatever code path the windows player is using? It's probably a stupid question, but I think I'm missing something :)

Reply:
I'm not sure I understand that last part about "special code path".  I don't think there is a special code path for windows player unless I'm gravely mistaken. Can @3iD0loN or @ekcoh correct me if I'm wrong.

Reply:
I'm having a little trouble following the explanation for why the virtual is need, but since I don't have a lot of exposure to either ecosystem, I'm going to have to rely on @ekcoh and @3iD0loN to steer me the right way here.

If you want the virtual gone, I'm happy to do it.  If it needs to stay, I'm happy to do that too.  @jamesmcgill, maybe you can break the tie here.

Reply:
There is no implementation for Players except AndroidPlayer. All Editors have this implemented in a way that the GameView when set top a display can simulate DisplayManager displays.

Before these changes, the displayIndex value did not change on the managed side, because it did not change in the unmanaged side, and so displayIndex was always 0 for all platforms. However, since we are now implementing it for Android, WinEditor, LinuxEditor, and MacEditor, as far as I'm aware we are now stuck with a discrepancy between platforms, where there is an expectation on AndroidPlayer that display index will be 1 when second screen is touched, but on say for instance WindowsPlayer it will remain 0, which may cause these asserts to fail.

And so the description of this PR is a bit incorrect and should be reworded to avoid further confusion.

As to why they are different values for different platforms, it because of how displayIndex is populated on the unmanaged side of code. The AndroidPlayer does this on the part where Android JNI Bridge interfaces with InputSystem, and for Editors, it retrieves the value from the GameView on the Editor's managed side and passes it towards the many platform-dependent structures. WindowsPlayer and others that are unimplemented wouyld handle population of displayIndex with their own native APIs. 
For deeper context down the avenue of platform dependent details, here are the changes that implement displayIndex population for Editors: https://ono.unity3d.com/unity/unity/pull-request/142411/_/ikin-ryz-input-system-display-index-binding-trunk
And for AndroidPlayer:
https://ono.unity3d.com/unity/unity/pull-request/134020/_/ikin-ryz-android-presentation-input-changes-trunk

Feel free to ask anything else as well.

Reply:
Thanks for elaborating on this. We decided to update this code to not have platform specific handling from Input System perspective, instead the displayIndex will be set to correct value when platform is supporting it and remain 0 when not supported. This drives a need to be very clear on what platforms are currently supporting this in user-communication / docs to not get a lot of bug reports which would basically be feature requests on platform capabilities.